add missing countries to countries list

npracht opened the pull request:
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | x
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
A country was missing in the country list.

npracht committed:
add missing french polynesia

npracht commented:
Label: Bug

npracht commented:
Label: Ready to test

npracht commented:
Label: Code Review

npracht committed:
bermuda where also missing

npracht committed:
add several missing countries

Add  Aruba
- Saint Pierre and Miquelon
- Guadeloupe
- Martinique
- New Caledonia
- Réunion

npracht commented:
Hello @escopecz is there any reason to do not include that into the 2.14?
Thanks !

galvani commented:
I am not sure how to handle this. I have found even more:
```
array(51) {
  ["WF"]=>
  string(17) "Wallis and Futuna"
  ["BL"]=>
  string(16) "Saint Barthelemy"
  ["BV"]=>
  string(13) "Bouvet Island"
  ["BQ"]=>
  string(34) "Bonaire, Saint Eustatius and Saba "
  ["JE"]=>
  string(6) "Jersey"
  ["RE"]=>
  string(7) "Reunion"
  ["TK"]=>
  string(7) "Tokelau"
  ["GW"]=>
  string(13) "Guinea-Bissau"
  ["GU"]=>
  string(4) "Guam"
  ["GS"]=>
  string(44) "South Georgia and the South Sandwich Islands"
  ["GG"]=>
  string(8) "Guernsey"
  ["GF"]=>
  string(13) "French Guiana"
  ["GL"]=>
  string(9) "Greenland"
  ["GI"]=>
  string(9) "Gibraltar"
  ["HK"]=>
  string(9) "Hong Kong"
  ["HM"]=>
  string(33) "Heard Island and McDonald Islands"
  ["PR"]=>
  string(11) "Puerto Rico" <=== this is for example a kind of colony, but French Guiana is too
  ["PS"]=>
  string(21) "Palestinian Territory"
  ["SJ"]=>
  string(22) "Svalbard and Jan Mayen"
  ["PN"]=>
  string(8) "Pitcairn"
  ["EH"]=>
  string(14) "Western Sahara"
  ["MF"]=>
  string(12) "Saint Martin"
  ["MO"]=>
  string(5) "Macao"
  ["MP"]=>
  string(24) "Northern Mariana Islands"
  ["MS"]=>
  string(10) "Montserrat"
```
and some current that may be replaced: I have checked just few with wikipedia.

```
array(5) {
  [73]=>
  string(13) "Guinea Bissau"
  [76]=>
  string(8) "Holy See"
  [145]=>
  string(8) "Réunion"
  [171]=>
  string(9) "Palestine"
  [192]=>
  string(7) "Unknown"
}
```

I would love to hear someone else's opinion about this.

galvani commented:
I recommend to compare the list to more sources and verify each record. PR like this is incomplete.

galvani reviewed (changes requested):
Provide complete list of missing Countries. Current changes seem not complete.

npracht commented:
Hello @galvani i can add it.
Question, shouldn't we set something more dynamic? I'm not a dev, but how far can we use https://symfony.com/doc/4.1/reference/constraints/Country.html in our case ?

galvani commented:
We cannot relay on external sources as they might change the list and we would loose data integrity. Same reason why we cannot rename standing list items.

npracht committed:
add some more missing countries

npracht commented:
Since `Puerto Rico` is in Despacito song i think we whouls include it (joke) ---> all kind of colony may be included otherwise we would also exclude Greenland (Denmark), etc.

Ok for your message concerning external sources.

Update done 💚

npracht commented:
@galvani concerning the list in error...

```json
array(5) {
  [73]=>
  string(13) "Guinea Bissau"
  [76]=>
  string(8) "Holy See"
  [145]=>
  string(8) "Réunion"
  [171]=>
  string(9) "Palestine"
  [192]=>
  string(7) "Unknown"
}
```

It would be awesome to modify it, but how to do?

npracht commented:
According to ``ISO 3166-1`, those are also missing:

- Cayman Islands
- Cook Islands
- Falkland Islands
- Niue
- Saint Helena, Ascension and Tristan da Cunha
- Turks and Caicos Islands
- United States Virgin Islands

npracht committed:
add more countries from ISO_3166-1 countries list

galvani commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -157,12 +192,13 @@
     "Spain",
     "Sri Lanka",
-    "Saint Vincent and the Grenadines",
-    "Palestine",
```
Don't change these, please keep the standing records untouched.

galvani reviewed (changes requested):
Just revert changes to the current records

galvani commented:
great you have chosen such a source of data.

npracht committed:
revert palestine

npracht commented:
@galvani i've reverted Palestine. I just changed the position of `Saint Vincent and the Grenadines` to keep sorted alphabeticaly. But it is written the same way ;)

galvani reviewed (approved):
Good job

npracht commented:
Label: Code Review

npracht commented:
Label: Pending test confirmation

npracht committed:
remove american samoa

that was already existing as samoa

npracht committed:
add last missing countries

npracht committed:
Update countries.json

alanhartless commented:
Many of these added back were removed with https://github.com/Webmecanik/mautic/commit/b8c43e54cb4a5dbb496d5c480327ff7c0556a2c5#diff-137555d225e4c5f919086c68d1781646 because they are not countries but territories or something else. Antartica? 

Are these territories typically listed as countries in other systems?

npracht commented:
Hi @alanhartless, hm.
![capture d ecran 2018-07-06 a 17 20 06](https://user-images.githubusercontent.com/14075239/42386825-e23e9304-8140-11e8-87dc-8fa2829b4647.png)

In this new PR we've just followed the international ISO norm, `ISO 3166-1`.
I think following this type of reference is the best option.

heathdutton commented:
IMO It's best to stick close to ISO 3166-1, because "countryness" is so hotly debated. I agree with this PR.

proffalken commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -31,13 +38,14 @@
     "China",
     "Colombia",
     "Comoros",
-    "Republic of the Congo",
```
https://en.wikipedia.org/wiki/ISO_3166-1 lists the following two states with "Congo" in their name:

   * Congo
   * Congo (Democratic Republic of the)

Is there a reason why we've removed any reference to either here?

proffalken commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -56,23 +64,33 @@
     "Eritrea",
     "Estonia",
     "Ethiopia",
+    "Falkland Islands",
```
ISO-3166-1 lists this as "Falkland Islands (Malvinas)" - I'm British, so refer to them as the Falkland Islands, but not having the "Malvinas" part on the end could upset any Argentinian users of the platform.

proffalken commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -56,23 +64,33 @@
     "Finland",
     "France",
+    "French Guiana",
+    "French Polynesia",
```
https://en.wikipedia.org/wiki/ISO_3166-1 also lists "French Southern Territories", but I'm not sure whether we should be including these here or not as they don't have a permanent population.

proffalken commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -82,8 +100,10 @@
     "Ireland",
     "Israel",
     "Italy",
+    "Ivory Coast",
```
Ivory Coast doesn't appear to be listed on https://en.wikipedia.org/wiki/ISO_3166-1 ?

proffalken commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -82,8 +100,10 @@
     "Japan",
+    "Jersey",
     "Jordan",
     "Kazakhstan",
```
We seem to be missing both Koreas ("Democratic People's Repulic of" AKA "North", and "Republic of" AKA "South")

proffalken commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -99,50 +119,68 @@
     "Liechtenstein",
```
Apparently there is also a Lao People's Democratic Republic, but we're missing that one.

proffalken commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -99,50 +119,68 @@
     "Nicaragua",
     "Niger",
     "Nigeria",
+    "Niue",
```
We're missing "Norfolk Island"

proffalken commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -99,50 +119,68 @@
     "Niger",
     "Nigeria",
+    "Niue",
     "North Korea",
```
Ah, North Korea is here, but that's not it's correct name (see note above in the 'K' section on the two Koreas)

proffalken commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -99,50 +119,68 @@
     "Oman",
     "Pakistan",
     "Palau",
+    "Palestine",
```
Technically, this should be "Palestine, State of"

proffalken commented on app/bundles/CoreBundle/Assets/json/countries.json:
```diff
@@ -157,29 +195,31 @@
+    "Turks and Caicos Islands",
     "Tuvalu",
     "United Kingdom",
     "United States",
```
The next line is "Unknown", is this because there are countries which haven't been discovered yet, or should it say "Not Listed" instead? ;)

(Sorry, British pedant over here ;) )

proffalken reviewed (commented):
A few queries about some of the names and some of the ones that are on the wikipedia article but not on this list, other than that it looks good.

npracht commented:
@proffalken & @heathdutton i agree with those comments but it has been mentioned before that i cannot change previously existing values. Otherwise the existing contacts with the old values would generate issues.

Any solution?

proffalken commented:
@npracht short of also writing some kind of mapping function (which is probably overkill here!), not really.

If the renaming causes backwards compatibility issues and there isn't a sensible way forward, I'm happy for this to proceed as it currently is and be merged.

npracht commented:
Thanks @proffalken for that feedback.
@heathdutton your final word ? :)

heathdutton commented:
The word is "GO" this is better than it was :)
